send/ad_mu

```diff
@@ -762,41 +791,50 @@ sub process_groups() {
 
 	foreach my $ad_entry (@ad_entries_groups) {
 		my $cn = $ad_entry->get_value('cn');
-		unless (exists $perun_entries_group_map{$cn}) {
+		unless (exists $perun_entries_group_map{$cn} || $ad_entry->get_value('extensionAttribute1') eq 'FALSE') {
 
 			# Prevent clearing main license group Employee,Student,Alumni !!
 			unless (($ad_entry->dn() eq $employeeDN) or ($ad_entry->dn() eq $studentDN) or ($ad_entry->dn() eq $alumniDN) or ($ad_entry->dn() eq $student2DN)){
 
 				# clear members
-				my @empty_members = ();
-				$ad_entry->replace(
-					'member' => \@empty_members
-				);
-
-				my $grace_period = $ad_entry->get_value('extensionAttribute2');
-				unless (defined $grace_period) {
-					$ad_entry->replace(
-						'extensionAttribute2' => $groups_grace_period_date
-					);
+				# load members of a group from AD
+				my @to_be_removed = load_group_members($ldap, $ad_entry->dn(), $filter_groups);
+				my $response_remove = remove_members_from_entry($ad_entry, \@to_be_removed);
+				unless ($response_remove == $SUCCESS) {
+					ldap_log($service_name, "Failed to remove all members from group during the group removal process: ".$ad_entry->dn());
+					$counter_group_not_emptied++;
 				} else {
-					my @grace_period_array = Decode_Date_EU($grace_period);
-					if(Delta_Days(@today, @grace_period_array) < 0) {
-						# set this attribute to FALSE
+					my $grace_period = $ad_entry->get_value('extensionAttribute2');
+					unless (defined $grace_period) {
 						$ad_entry->replace(
-							'extensionAttribute1' => 'FALSE'
+							'extensionAttribute2' => $groups_grace_period_date
```

Comment:
These two attributes should be set as constants and you should add info to comments what does they mean. For example what values could be saved here and what exactly means to have TRUE or FALSE for such attribute set.

Reply:
Done